src/circuit/gadget/ecc/chip.rs

```diff
@@ -0,0 +1,361 @@
+    pub q_mul_complete: Selector,
+    /// Fixed-base full-width scalar multiplication
+    pub q_mul_fixed: Selector,
+    /// Fixed-base signed short scalar multiplication
+    pub q_mul_fixed_short: Selector,
+    /// Witness point
+    pub q_point: Selector,
+    /// Witness full-width scalar for fixed-base scalar mul
+    pub q_scalar_fixed: Selector,
+    /// Witness signed short scalar for full-width fixed-base scalar mul
+    pub q_scalar_fixed_short: Selector,
+    /// Permutation
+    pub perm: Permutation,
+    _marker: PhantomData<C>,
+}
+
+/// A chip implementing EccInstructions
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct EccChip<C: CurveAffine> {
+    config: EccConfig<C>,
+}
+
+impl<C: CurveAffine> Chip<C::Base> for EccChip<C> {
+    type Config = EccConfig<C>;
+    type Loaded = ();
+
+    fn config(&self) -> &Self::Config {
+        &self.config
+    }
+
+    fn loaded(&self) -> &Self::Loaded {
+        &()
+    }
+}
+
+impl<C: CurveAffine> EccChip<C>
+where
+    C::Scalar: PrimeFieldBits,
+{
+    pub fn construct(config: <Self as Chip<C::Base>>::Config) -> Self {
+        Self { config }
+    }
+
+    #[allow(non_snake_case)]
+    pub fn configure(
+        meta: &mut ConstraintSystem<C::Base>,
+        advices: [Column<Advice>; 10],
+        perm: Permutation,
+    ) -> <Self as Chip<C::Base>>::Config {
+        let config = EccConfig::<C> {
+            advices,
+            lagrange_coeffs: [
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+                meta.fixed_column(),
+            ],
+            fixed_z: meta.fixed_column(),
+            q_add_incomplete: meta.selector(),
+            q_add: meta.selector(),
+            q_mul_hi: meta.selector(),
+            q_mul_lo: meta.selector(),
+            q_mul_decompose_var: meta.selector(),
+            q_init_z: meta.selector(),
+            q_mul_complete: meta.selector(),
+            q_mul_fixed: meta.selector(),
+            q_mul_fixed_short: meta.selector(),
+            q_point: meta.selector(),
+            q_scalar_fixed: meta.selector(),
+            q_scalar_fixed_short: meta.selector(),
+            perm,
+            _marker: PhantomData,
+        };
+
+        // Create witness point gate
+        {
+            let config: witness_point::Config<C> = (&config).into();
+            config.create_gate(meta);
+        }
+
+        // Create incomplete point addition gate
+        {
+            let config: add_incomplete::Config<C> = (&config).into();
+            config.create_gate(meta);
+        }
+
+        // Create complete point addition gate
+        {
+            let add_config: add::Config<C> = (&config).into();
+            add_config.create_gate(meta);
+        }
+
+        // Create variable-base scalar mul gates
```

Comment:
```suggestion
        // Create variable-base scalar mul gate
```

Reply:
These documentation fixes have been addressed in c98b15239e82142f3a016da1d0495b1cd703fa24 on #111.